pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -570,41 +570,41 @@ func (p *staticPolicy) validateState(s state.State) error {
 
 func areMachineStatesEqual(ms1, ms2 state.NUMANodeMap) bool {
 	if len(ms1) != len(ms2) {
-		klog.Errorf("[memorymanager] node states are different len(ms1) != len(ms2): %d != %d", len(ms1), len(ms2))
+		klog.ErrorS(nil,"[memorymanager] Node states are different.", "len(ms1)", len(ms1), "len(ms2)", len(ms2))
 		return false
 	}
 
 	for nodeID, nodeState1 := range ms1 {
 		nodeState2, ok := ms2[nodeID]
 		if !ok {
-			klog.Errorf("[memorymanager] node state does not have node ID %d", nodeID)
+			klog.ErrorS(nil, "[memorymanager] Node state does not have node ID.", "nodeID", nodeID)
 			return false
 		}
 
 		if nodeState1.NumberOfAssignments != nodeState2.NumberOfAssignments {
-			klog.Errorf("[memorymanager] node states number of assignments are different: %d != %d", nodeState1.NumberOfAssignments, nodeState2.NumberOfAssignments)
+			klog.ErrorS(nil,"[memorymanager] Node states number of assignments are different.", "state1.NumberOfAssignments", nodeState1.NumberOfAssignments, "state2.NumberOfAssignments", nodeState2.NumberOfAssignments)
 			return false
 		}
 
 		if !areGroupsEqual(nodeState1.Cells, nodeState2.Cells) {
-			klog.Errorf("[memorymanager] node states groups are different: %v != %v", nodeState1.Cells, nodeState2.Cells)
+			klog.ErrorS(nil, "[memorymanager] Node states groups are different.", "state1.Cells", nodeState1.Cells, "state2.Cells", nodeState2.Cells)
 			return false
 		}
 
 		if len(nodeState1.MemoryMap) != len(nodeState2.MemoryMap) {
-			klog.Errorf("[memorymanager] node states memory map have different length: %d != %d", len(nodeState1.MemoryMap), len(nodeState2.MemoryMap))
+			klog.ErrorS(nil, "[memorymanager] Node states memory map have different length.", "len(state1.MemoryMap)", len(nodeState1.MemoryMap), "len(state2.MemoryMap)", len(nodeState2.MemoryMap))
```

Comment:
```suggestion
			klog.ErrorS(nil, "[memorymanager] Node states memory map have different length", "len1", len(nodeState1.MemoryMap), "len2", len(nodeState2.MemoryMap))
```

Reply:
not sure here, keys could be `state1` or `state2` or `lengthState1` `lengthState2`